glusterd2/commands/volumes/volume-profile.go

```diff
@@ -0,0 +1,312 @@
+package volumecommands
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/gluster/glusterd2/glusterd2/brick"
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/servers/sunrpc/dict"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+
+	"github.com/gorilla/mux"
+)
+
+func registerVolProfileStepFuncs() {
+	transaction.RegisterStepFunc(txnVolumeProfile, "volume.Profile")
+}
+
+func volumeProfileHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+	volname := mux.Vars(r)["volname"]
+	option := mux.Vars(r)["option"]
+
+	txn, err := transaction.NewTxnWithLocks(ctx, volname)
```

Comment:
can we first check volume is present in glusterd2? 
 if the volume is present then take a lock on it.

Reply:
Asked @prashanthpai about this issue. He says that its better to check for volinfo after the locks have been taken.

Reply:
my only concern is, if we get a large number of requests, we should not end up in exhausting ETCD locks.